src/auth/auth.service.ts

```diff
@@ -34,35 +34,56 @@ export class AuthService {
         }
     }
 
-    GetAllUsers(): Promise<User[]> {
-        const user = this.userRepository.find()
+    async GetAllUsers(page, size): Promise<any> {
+
+        const user = await this.userRepository.find({
+            take: size,
+            skip: size * (page - 1)
+        })
+
+        if (user.length === 0) {
+            throw new UnauthorizedException(`end of the page`)
+        }
+
         return user
     }
 
 
-    async searchByUsername(authCredDto: AuthCredDto, page): Promise<any> {
+    async searchByUsername(authCredDto: AuthCredDto, page, page_size): Promise<any> {
         const { user_name } = authCredDto
 
-        const user = await this.userRepository.find({ user_name })
+        const user = await this.userRepository.find(
+            {
+                where: {
+                    user_name: user_name,
+                },
+                take: page_size,
+                skip: page_size * (page - 1)
+            }
+        )
 
-        console.log(user.length)
+        // console.log(user)
+        if (user.length === 0) {
+            throw new UnauthorizedException(`End of The Page`)
+        }
 
-        if (user.length > 1) {
-            let page_size = 2;
+        return user
 
-            let temp;
 
-            function paginate(array, page_size, page_number) {
+        // if (user.length > 1) {
+        //     let temp;
 
-                return array.slice((page_number - 1) * page_size, page_number * page_size);
-            }
-            temp = paginate(user, page_size, page);
+        //     function paginate(array, page_size, page_number) {
 
-            console.log(user.length)
-            return temp
-        } else {
-            throw new UnauthorizedException(`couldn't found the username ${user_name}`)
-        }
+        //         return array.slice((page_number - 1) * page_size, page_number * page_size);
+        //     }
+        //     temp = paginate(user, page_size, page);
+
+        //     // console.log(user.length)
```

Comment:
Remove unsed codes and console.log

Reply:
done with removing unused codes and console.log

Reply:
Removed unused codes and console.log